### ISSUE TITLE
Cannot load frontend for Tutorial: Build a proof of existence dApp

### ISSUE THREAD

**Issue by teamchong** [substrate-developer-hub/substrate-docs#836]
### Bug report for the website

I followed the instruction here.
[Tutorial: Build a proof of existence dApp | Substrate_](https://docs.substrate.io/tutorials/v3/proof-of-existence/)
The node-template works fine.
But the frontend starting page (http://localhost:8000/substrate-front-end-template) cannot be loaded.

To fix the issue, I have to replace `const { api } = useSubstrate()` with `const { api } = useSubstrateState()`

[https://github.com/teamchong/substrate-proof-of-existence-frontend/commit/e138b605857ccbe7085da4d4ce5ff372230555ab](https://github.com/teamchong/substrate-proof-of-existence-frontend/commit/e138b605857ccbe7085da4d4ce5ff372230555ab)

### Steps to reproduce the problem

Following is copied from [https://docs.substrate.io/tutorials/v3/proof-of-existence/](https://docs.substrate.io/tutorials/v3/proof-of-existence/)
1. Open a new terminal shell on your computer, then change to the root directory where you installed the front-end template.
2. Open the `src/TemplateModule.js` file in a text editor.
3. Delete the entire contents of that file.
4. Copy and paste the following code into the `src/TemplateModule.js` file:
```js
// React and Semantic UI elements.
import React, { useState, useEffect } from 'react'
import { Form, Input, Grid, Message } from 'semantic-ui-react'

// Pre-built Substrate front-end utilities for connecting to a node
// and making a transaction.
import { useSubstrate } from './substrate-lib'
import { TxButton } from './substrate-lib/components'

// Polkadot-JS utilities for hashing data.
import { blake2AsHex } from '@polkadot/util-crypto'

// Main Proof Of Existence component is exported.
export function Main(props) {
  // Establish an API to talk to the Substrate node.
  const { api } = useSubstrate()
  // Get the selected user from the `AccountSelector` component.
  const { accountPair } = props
  // React hooks for all the state variables we track.
  // Learn more at: https://reactjs.org/docs/hooks-intro.html
  const [status, setStatus] = useState('')
  const [digest, setDigest] = useState('')
  const [owner, setOwner] = useState('')
  const [block, setBlock] = useState(0)
  // Our `FileReader()` which is accessible from our functions below.
  let fileReader
  // Takes our file, and creates a digest using the Blake2 256 hash function
  const bufferToDigest = () => {
    // Turns the file content to a hexadecimal representation.
    const content = Array.from(new Uint8Array(fileReader.result))
      .map(b => b.toString(16).padStart(2, '0'))
      .join('')
    const hash = blake2AsHex(content, 256)
    setDigest(hash)
  }

  // Callback function for when a new file is selected.
  const handleFileChosen = file => {
    fileReader = new FileReader()
    fileReader.onloadend = bufferToDigest
    fileReader.readAsArrayBuffer(file)
  }

  // React hook to update the owner and block number information for a file
  useEffect(() => {
    let unsubscribe
    // Polkadot-JS API query to the `proofs` storage item in our pallet.
    // This is a subscription, so it will always get the latest value,
    // even if it changes.
    api.query.templateModule
      .proofs(digest, result => {
        // Our storage item returns a tuple, which is represented as an array.
        setOwner(result[0].toString())
        setBlock(result[1].toNumber())
      })
      .then(unsub => {
        unsubscribe = unsub
      })
    return () => unsubscribe && unsubscribe()
    // This tells the React hook to update whenever the file digest changes
    // (when a new file is chosen), or when the storage subscription says the
    // value of the storage item has updated.
  }, [digest, api.query.templateModule])

  // We can say a file digest is claimed if the stored block number is not 0
  function isClaimed() {
    return block !== 0
  }

  // The actual UI elements which are returned from our component.
  return (
    <Grid.Column>
      <h1>Proof of Existence</h1>
      {/* Show warning or success message if the file is or is not claimed. */}
      <Form success={!!digest && !isClaimed()} warning={isClaimed()}>
        <Form.Field>
          {/* File selector with a callback to `handleFileChosen`. */}
          <Input
            type="file"
            id="file"
            label="Your File"
            onChange={e => handleFileChosen(e.target.files[0])}
          />
          {/* Show this message if the file is available to be claimed */}
          <Message success header="File Digest Unclaimed" content={digest} />
          {/* Show this message if the file is already claimed. */}
          <Message
            warning
            header="File Digest Claimed"
            list={[digest, `Owner: ${owner}`, `Block: ${block}`]}
          />
        </Form.Field>
        {/* Buttons for interacting with the component. */}
        <Form.Field>
          {/* Button to create a claim. Only active if a file is selected, and not already claimed. Updates the `status`. */}
          <TxButton
            accountPair={accountPair}
            label={'Create Claim'}
            setStatus={setStatus}
            type="SIGNED-TX"
            disabled={isClaimed() || !digest}
            attrs={{
              palletRpc: 'templateModule',
              callable: 'createClaim',
              inputParams: [digest],
              paramFields: [true],
            }}
          />
          {/* Button to revoke a claim. Only active if a file is selected, and is already claimed. Updates the `status`. */}
          <TxButton
            accountPair={accountPair}
            label="Revoke Claim"
            setStatus={setStatus}
            type="SIGNED-TX"
            disabled={!isClaimed() || owner !== accountPair.address}
            attrs={{
              palletRpc: 'templateModule',
              callable: 'revokeClaim',
              inputParams: [digest],
              paramFields: [true],
            }}
          />
        </Form.Field>
        {/* Status message about the transaction. */}
        <div style={{ overflowWrap: 'break-word' }}>{status}</div>
      </Form>
    </Grid.Column>
  )
}

export default function TemplateModule(props) {
  const { api } = useSubstrate()
  return api.query.templateModule && api.query.templateModule.proofs ? (
    <Main {...props} />
  ) : null
}
```
5. Save your changes and close the file.
6. Start the front-end template by running the following command:
    `yarn start`

This will open up a new tab with the front-end serving at http://localhost:8000.

**Comment by teamchong** [comment]
Blank page after select a file and "Create Claim".
there is no `props.accountPair`
and this line `disabled={!isClaimed() || owner !== accountPair.address}` will cause an error cause accountPair is undefined.

To fix the issue [https://github.com/teamchong/substrate-proof-of-existence-frontend/commit/ca16cf786ecf96bdb635286e14ea3344c9e24342](https://github.com/teamchong/substrate-proof-of-existence-frontend/commit/ca16cf786ecf96bdb635286e14ea3344c9e24342)

**Comment by NukeManDan** [comment]
Resolved in #861
